contrib/storage-hive/core/src/main/java/org/apache/drill/exec/store/hive/HiveRecordReader.java

```diff
@@ -223,6 +228,24 @@ private void init() throws ExecutionSetupException {
   @Override
   public void setup(@SuppressWarnings("unused") OperatorContext context, OutputMutator output)
       throws ExecutionSetupException {
+    final ListenableFuture<Void> result = context.runCallableAs(proxyUgi,
+      new Callable<Void>() {
+        @Override
+        public Void call() throws Exception {
+          init();
+          return null;
+        }
+      });
+    try {
+      result.get();
+    } catch (InterruptedException e) {
+      result.cancel(true);
+      // Preserve evidence that the interruption occurred so that code higher up on the call stack can learn of the
+      // interruption and respond to it if it wants to.
+      Thread.currentThread().interrupt();
```

Comment:
I don't think you want to set the interrupt bit on the proxy thread, you probably need throw the interrupted exception in the original context.  I also suggest creating a little inner class rather than an anonymous object for easier understanding.